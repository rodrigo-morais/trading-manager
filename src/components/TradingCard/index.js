import React from 'react'
import styled from 'styled-components'

import Header from '../Header'
import Start from '../Start'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(8, 1fr);
  grid-row-gap: 8%;
  justify-content: space-evenly;
  width: 49%;
  border: solid 1px #000000;
  border-radius: 12px;
  background-color: #AACAA2;
  padding: 0.5%;
`

const FirstLine = styled(Header)`
  grid-row-start: 1;
  grid-column-start: 1;
  grid-column-end: 1;
  justify-self: start;
`

const SecondLine = styled(Start)`
  grid-row-start: 2;
  grid-column-start: 1;
  grid-column-end: 1;
  justify-self: start;
`

const TradingCard = () => (
  <Container>
    <FirstLine />
    <SecondLine />
  </Container>
)

export default TradingCard
