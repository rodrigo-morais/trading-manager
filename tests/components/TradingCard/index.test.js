import React from 'react'
import renderer from 'react-test-renderer'

import TradingCard from '../../../src/components/TradingCard'

describe('TradingCard', () => {
  describe('Rendering', () => {
    const wrapper = (
      <TradingCard />
    )

    test('renders according to design', () => {
      const component = renderer.create(wrapper)
      const tree = component.toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
