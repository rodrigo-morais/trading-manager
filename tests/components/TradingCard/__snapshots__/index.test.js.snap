// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradingCard Rendering renders according to design 1`] = `
<div
  className="sc-dnqmqq UNfzD"
>
  <div
    className="sc-iwsKbI dpxXYA sc-bxivhb hmJKfe"
  >
    <div
      className="sc-bdVaJa lhrPED"
    >
      <label
        className="sc-bwzfXH ebnOHd"
        htmlFor="Ativo"
      >
        Ativo:
      </label>
      <select
        className="sc-htpNat eSoGCX"
        id="Ativo"
        name="Ativo"
      >
        <option
          value="WINM20"
        >
          WINM20
        </option>
        <option
          value="WDON20"
        >
          WDON20
        </option>
      </select>
    </div>
    <div
      className="sc-bdVaJa lhrPED"
    >
      <label
        className="sc-bwzfXH ebnOHd"
        htmlFor="Tipo"
      >
        Tipo:
      </label>
      <select
        className="sc-htpNat eSoGCX"
        id="Tipo"
        name="Tipo"
      >
        <option
          value="Pregão"
        >
          Pregão
        </option>
        <option
          value="Abertura"
        >
          Abertura
        </option>
      </select>
    </div>
    <div
      className="sc-bdVaJa lhrPED"
    >
      <label
        className="sc-bwzfXH ebnOHd"
        htmlFor="Estratégia"
      >
        Estratégia:
      </label>
      <select
        className="sc-htpNat eSoGCX"
        id="Estratégia"
        name="Estratégia"
      >
        <option
          value="Topo / Fundo"
        >
          Topo / Fundo
        </option>
        <option
          value="Fecha Fora / Fecha Dentro"
        >
          Fecha Fora / Fecha Dentro
        </option>
        <option
          value="3x1"
        >
          3x1
        </option>
        <option
          value="Fabolous 4"
        >
          Fabolous 4
        </option>
      </select>
    </div>
  </div>
  <div
    className="sc-gZMcBi kDoKwg sc-gzVnrw dvLPbt"
  >
    <div
      className="sc-ifAKCX klasqr"
    >
      <label
        className="sc-EHOje eNQsqv"
        htmlFor="Entrada"
      >
        Entrada:
      </label>
      <input
        className="sc-bZQynM cdAyGJ"
        disabled={false}
        id="Entrada"
        name="Entrada"
        type="text"
      />
    </div>
    <div
      className="sc-ifAKCX klasqr"
    >
      <label
        className="sc-EHOje eNQsqv"
        htmlFor="Stop Base"
      >
        Stop Base:
      </label>
      <input
        className="sc-bZQynM cdAyGJ"
        disabled={false}
        id="Stop Base"
        name="Stop Base"
        type="text"
      />
    </div>
    <div
      className="sc-ifAKCX klasqr"
    >
      <label
        className="sc-EHOje eNQsqv"
        htmlFor="Hora"
      >
        Hora:
      </label>
      <input
        className="sc-bZQynM cdAyGJ"
        disabled={false}
        id="Hora"
        name="Hora"
        type="text"
      />
    </div>
    <button
      className="sc-htoDjs jLAlE"
      disabled={false}
      value="Iniciar"
    >
      Iniciar
    </button>
  </div>
</div>
`;
